data-plane/dispatcher-loom/src/test/java/dev/knative/eventing/kafka/broker/dispatcherloom/LoomKafkaConsumerTest.java

```diff
@@ -0,0 +1,190 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.knative.eventing.kafka.broker.dispatcherloom;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import io.vertx.core.Vertx;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.MockConsumer;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.clients.consumer.OffsetResetStrategy;
+import org.apache.kafka.common.TopicPartition;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(VertxExtension.class)
+public class LoomKafkaConsumerTest {
+
+    private Vertx vertx;
+    private LoomKafkaConsumer<String, Integer> consumer;
+    private MockConsumer<String, Integer> mockConsumer;
+
+    @BeforeEach
+    public void setUp() {
+        vertx = Vertx.vertx();
+        mockConsumer = new MockConsumer<String, Integer>(OffsetResetStrategy.LATEST);
+        consumer = new LoomKafkaConsumer<String, Integer>(vertx, mockConsumer);
+    }
+
+    @AfterEach
+    public void tearDown() {
+        consumer.close();
+    }
+
+    @Test
+    public void testTaskRunnerThreadIsStopAfterClose(VertxTestContext testContext) throws InterruptedException {
+
+        final var checkpoints = testContext.checkpoint();
+
+        // add task to TaskRunnerThread
+        consumer.subscribe(Collections.singletonList("test-topic"));
+        consumer.poll(Duration.ofMillis(1000));
+        consumer.commit(new HashMap<>());
+
+        // Verify that the task runner thread is running
+        assertTrue(consumer.isTaskRunnerThreadAlive());
+
+        // Close the consumer
+        consumer.close().onFailure(testContext::failNow);
+
+        // Verify that the task runner thread is stopped after 1 second
+        Thread.sleep(1000L);
+        assertFalse(consumer.isTaskRunnerThreadAlive());
+        checkpoints.flag();
```

Comment:
I would place this inside the `consumer.close().onComplete( <here> )` without the `thread.sleep`

Reply:
Actually, the future of close function doesn't ensure that the thread has stopped.
There is a completely different thread just to stop the taskRunnerThread.

So I have tried `onComplete(...)` but in that case, it is failing very occasionally, almost 1 out of 15 or 20 times. So I go with this approach.